### PR TITLE
First pull request

### PR DESCRIPTION
**Pull request by abcussi**
- test the methods with 4 examples for each method.
- the methods pass all the tests.
- using procs' in the injects test.
- using symbols in the inject test.
- testing methods with Booleans', numbers and strings.

### CODE REVIEW

File: spec/enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  end
+
+  def test_4_my_inject
+    %w[hello world my name is].my_inject(false) { |total, x| total + ' ' + x }
+  end
+end
+
+RSpec.describe Enum do
+  describe '#add' do
+    enum = Enum.new
```
Comment (suretrust) [inline spec/enum_spec.rb]: Use `let` here instead. You can check [this article](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) to understand why.

File: spec/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    it 'testing method my_each' do
+      expect(enum.test_1_my_each).to eql(2)
```
Comment (suretrust) [inline spec/enum_spec.rb:8]: * [ ] Ensure to test for negative scenarios. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
#### Example
```ruby
expect(enum.test_1_my_each).to eql(2)
expect(enum.test_1_my_each).not_to eql(4)
```

File: spec/enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      expect(enum.test_2_my_each).to eql('hello hi world oscar angel ')
+      expect(enum.test_3_my_each).to eql(false)
+      expect(enum.test_4_my_each).to eql('hi')
+    end
+    it 'testing method my_each_with_index' do
```
Comment (suretrust) [inline spec/enum_spec.rb]: `[Best practice][Optional]:` When writing your `it` block, it should be followed by a singular verb.
#### Example:
```ruby
it 'returns the sum of odd values' do
   expect(enum.test_1_my_each_with_index).to eql(8)
   expect(enum.test_1_my_each_with_index).not_to eql(13)
end

it 'returns the sum of even values' do
   expect(enum.test_2_my_each_with_index).to eql(11)
   expect(enum.test_2_my_each_with_index).not_to eql(16)
end
```

File: spec/enum_spec.rb [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    counter = 0
+    [1, 2, 3, 6, 7].my_each { |x| counter = x.even? ? counter + 1 : counter }
+    counter
+  end
+
+  def test_2_my_each
+    concat = ''
+    %w[hello hi world oscar angel].my_each { |x| concat += x + ' ' }
+    concat
+  end
```
Comment (suretrust) [inline spec/enum_spec.rb]: `[Optional]`: You could move the `Enum` class to a new file and import it into this file using `require_relative.` This makes your code more modular.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job implementing the tests! 🙌 Please see my comments on your code. 👇 

**Kindly submit another code review request once this has been fixed.** _Happy coding!_ 🎉

**Comment by andresporras3423** [comment]
* Using let for enum instance
* testing negative scenarios with not_to eql
*  Using singular verb for it blocks
* enum class is in another file

File: spec/enum_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      expect(enum.test_1_my_each).to eql(2)
+      expect(enum.test_2_my_each).to eql('hello hi world oscar angel ')
+      expect(enum.test_3_my_each).not_to eql(true)
+      expect(enum.test_4_my_each).to eql('hi')
+    end
+    it 'tests method my_each_with_index' do
+      expect(enum.test_1_my_each_with_index).to eql(8)
+      expect(enum.test_2_my_each_with_index).to eql(11)
+      expect(enum.test_3_my_each_with_index).not_to eql(false)
+      expect(enum.test_4_my_each_with_index).to eql('hello world')
```
Comment (euqueme) [inline spec/enum_spec.rb]: - [ ] You should have one `it` block for each expected behavior [for example:](https://github.com/rubocop-hq/rspec-style-guide#let-blocks)

- [ ] Also be more descriptive with every behaviour :point_down: 
```
.
.
describe "my_each_with_index" do
  it "handles a variable within a block and display its result" do
    expect(enum.test_2_my_each_with_index).to eql(11)
  end 
  it "handles a logic variable within a block and it doesn't display a wrongful result" do
   expect(enum.test_3_my_each_with_index).not_to eql(false)
  end 
.
.
.
end
```
please do this for all your test cases not just this one :v:

File: spec/enum_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
+require_relative '../enumerable_methods'
+require_relative '../enum'
+
+RSpec.describe Enum do
+  describe '#add' do
```
Comment (euqueme) [inline spec/enum_spec.rb]: - [ ] Enclose all test cases from each enumerable method  in their own `describe` block [for example](https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples):

```
Rspec.describe Enum do
let(:enum) { Enum.new }
describe 'my_each' do
  it 'test case # 1' do
  end
  it 'test case # 2' do
  end
end
.
.
.
describe 'my_map' do
  it 'test case # 1' do
  end
  it 'test case # 2' do
  end
end
```

do this for all your enum methods :v:

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @abcussi and @andresporras3423 :handshake: 

### Thank you for doing the changes requested by the previous reviewer, :medal_sports: your project looks better now still as a follow up of the changes you've made I've requested some changes in your code bellow, please address them :point_down: 

### that's it, team! :raised_hands: 

### Please :pray:  do the changes :man_technologist: :man_technologist: and submit back for a code review when you are done

Happy coding! :rocket: 

Any question? tell me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

**Comment by andresporras3423** [comment]
Hi Maria Eugina, 

Thank you for your feedback, I will make the required changes.

**Comment by andresporras3423** [comment]
* Using one 'it' for each test
* One 'describe' for each method
* More descriptive with the purpose of each test

**Review by whiz25 — APPROVED** [review]
## Congz :clap: 
Your code looks good and all test cases pass. It's a :+1: from me. Merge and proceed.

### Martin Afani

**Comment by andresporras3423** [comment]
Thank you Afani

**Pull request merged**